programs/openbook-v2/src/state/orderbook/book.rs

```diff
@@ -200,8 +210,34 @@ impl<'a> Orderbook<'a> {
                 best_opposing_price,
                 match_base_lots,
             );
-            event_queue.push_back(cast(fill)).unwrap();
+
+            let loader = remaining_accs.iter().find(|ai| ai.key == &fill.maker);
+            if let Some(acc) = loader {
+                let ooa: AccountLoader<OpenOrdersAccountFixed> = AccountLoader::try_from(acc)?;
+                let mut maker = ooa.load_full_mut()?;
+
+                maker.execute_maker(market, &fill)?;
+                emit!(FillLog {
```

Comment:
you could move `FillLog` inside `execute_maker` to avoid dup code here & in the `consume_events`